--- halt.orig	2008-06-05 21:09:31.453907988 +0200
+++ halt	2008-06-05 22:25:35.339146789 +0200
@@ -8,6 +8,21 @@
 #               Modified for RHS Linux by Damien Neil
 #
 
+# Patched for comoonics patch 1.3
+
+
+COM_KILLALL_OPTS="-x aisexec \
+-x ccsd \
+-x fenced \
+-x gfs_controld \
+-x dlm_controld \
+-x groupd \
+-x qdiskd \
+-x clvmd"
+
+COM_CHROOT_PATH=$(/opt/atix/comoonics-bootimage/manage_chroot.sh -p | sed s#/#\\\\\/#g)
+COM_CHROOT_PATH_REAL=$(/opt/atix/comoonics-bootimage/manage_chroot.sh -p)
+
 NOLOCALE=1
 . /etc/init.d/functions
 
@@ -65,9 +80,11 @@
 # Kill all processes.
 [ "${BASH+bash}" = bash ] && enable kill
 
-action $"Sending all processes the TERM signal..." /sbin/killall5 -15
+action $"Sending all processes the TERM signal..." /usr/comoonics/sbin/killall5 -15 $COM_KILLALL_OPTS 
 sleep 5
-action $"Sending all processes the KILL signal..."  /sbin/killall5 -9
+action $"Sending all processes the KILL signal..."  /usr/comoonics/sbin/killall5 -9 $COM_KILLALL_OPTS 
+
+/sbin/ifup lo &> /dev/null
 
 # Write to wtmp file before unmounting /var
 /sbin/halt -w
@@ -129,7 +146,9 @@
 [ -x /sbin/hwclock ] && action $"Syncing hardware clock to system time" /sbin/hwclock $CLOCKFLAGS
 
 # Try to unmount tmpfs filesystems to avoid swapping them in.  Ignore failures.
-tmpfs=$(awk '$2 ~ /^\/($|proc|dev)/ { next; }
+tmpfs=$(awk '
+$2 ~ /'$COM_CHROOT_PATH'/ || $2 ~/^\/cdsl.local/ {next}
+$2 ~ /^\/($|proc|dev)/ { next; }
 	     $3 == "tmpfs" { print $2; }' /proc/mounts | sort -r)
 [ -n "$tmpfs" ] && fstab-decode umount $tmpfs 2>/dev/null
 
@@ -163,6 +182,7 @@
     -f
 
 LANG=C __umount_loop '$2 ~ /^\/$|^\/proc|^\/dev/{next}
+	$2 ~ /'$COM_CHROOT_PATH'/ || $2 ~/^\/cdsl.local/ {next}
 	$3 == "tmpfs" || $3 == "proc" {print $2 ; next}
 	/(loopfs|autofs|nfs|cifs|smbfs|ncpfs|sysfs|^none|^\/dev\/ram|^\/dev\/root$)/ {next}
 	{print $2}' /proc/mounts \
@@ -180,7 +200,9 @@
 
 # Try all file systems other than root, essential filesystems and RAM disks,
 # one last time.
-awk '$2 !~ /\/(|dev|proc|selinux)$/ && $1 !~ /^\/dev\/ram/ { print $2 }' \
+awk '
+$2 ~ /'$COM_CHROOT_PATH'/ || $2 ~/^\/cdsl.local/ {next}
+$2 !~ /\/(|dev|proc|selinux)$/ && $1 !~ /^\/dev\/ram/ { print $2 }' \
     /proc/mounts | sort -r | \
   while read line; do
     fstab-decode umount -f $line
@@ -192,9 +214,9 @@
 
 # Remount read only anything that's left mounted.
 # echo $"Remounting remaining filesystems readonly"
-mount | awk '{ print $3 }' | while read line; do
-    fstab-decode mount -n -o ro,remount $line
-done
+#mount | awk '{ print $3 }' | while read line; do
+#fstab-decode mount -n -o ro,remount $line
+#done
 
 # Now halt or reboot.
 echo $"$message"
@@ -219,4 +241,5 @@
 HALTARGS="-d"
 [ -f /poweroff -o ! -f /halt ] && HALTARGS="$HALTARGS -p"
 
+$COM_CHROOT_PATH_REAL/com-halt.sh "$command $HALTARGS"
 exec $command $HALTARGS
