--- netfs.orig	2011-07-28 11:29:38.000000000 +0200
+++ netfs	2011-07-28 11:48:01.359284813 +0200
@@ -17,20 +17,22 @@
 . /etc/init.d/functions
 . /etc/sysconfig/network
 
+xtab=$(awk 'BEGIN{printf"/("}{ gsub(/\//, "\\/"); printf $0"|";}END{printf"^$)/"}' /etc/xtab 2>/dev/null)
+
 # Check that networking is up.
 [ "${NETWORKING}" = "no" ] && exit 0
 
-NFSFSTAB=`LC_ALL=C awk '!/^#/ && $3 ~ /^nfs/ && $3 != "nfsd" && $4 !~ /noauto/ { print $2 }' /etc/fstab`
+NFSFSTAB=`LC_ALL=C awk '!/^#/ && $3 ~ /^nfs/ && $3 != "nfsd" && $4 !~ /noauto/ && $2 !~ '$xtab' { print $2 }' /etc/fstab`
 SMBFSTAB=`LC_ALL=C awk '!/^#/ && $3 == "smbfs" && $4 !~ /noauto/ { print $2 }' /etc/fstab`
 CIFSFSTAB=`LC_ALL=C awk '!/^#/ && $3 == "cifs" && $4 !~ /noauto/ { print $2 }' /etc/fstab`
 NCPFSTAB=`LC_ALL=C awk '!/^#/ && $3 == "ncpfs" && $4 !~ /noauto/ { print $2 }' /etc/fstab`
-NETDEVFSTAB=`LC_ALL=C awk '!/^#/ && $4 ~/_netdev/ && $4 !~ /noauto/ { print $1 }' /etc/fstab`
+NETDEVFSTAB=`LC_ALL=C awk '!/^#/ && $4 ~/_netdev/ && $4 !~ /noauto/ && $2 !~ '$xtab' { print $1 }' /etc/fstab`
 
-NFSMTAB=`LC_ALL=C awk '$3 ~ /^nfs/ && $3 != "nfsd" && $2 != "/" { print $2 }' /proc/mounts`
+NFSMTAB=`LC_ALL=C awk '$3 ~ /^nfs/ && $3 != "nfsd" && $2 != "/" && $2 !~ '$xtab' { print $2 }' /proc/mounts`
 SMBMTAB=`LC_ALL=C awk '$3 == "smbfs" { print $2 }' /proc/mounts`
 CIFSMTAB=`LC_ALL=C awk '$3 == "cifs" { print $2 }' /proc/mounts`
 NCPMTAB=`LC_ALL=C awk '$3 == "ncpfs" { print $2 }' /proc/mounts`
-NETDEVMTAB=`LC_ALL=C awk '$4 ~ /_netdev/ && $2 != "/" { print $2 }' /etc/mtab`
+NETDEVMTAB=`LC_ALL=C awk '$4 ~ /_netdev/ && $2 != "/" && $2 !~ '$xtab' { print $2 }' /etc/mtab`
 
 # See how we were called.
 case "$1" in
@@ -104,13 +106,13 @@
         # Unmount loopback stuff first
 	__umount_loopback_loop
   	if [ -n "$NETDEVMTAB" ]; then
-		__umount_loop '$4 ~ /_netdev/ && $2 != "/" {print $2}' \
+		__umount_loop '$4 ~ /_netdev/ && $2 != "/" && $2 !~ '$xtab' {print $2}' \
 			/etc/mtab \
 			$"Unmounting network block filesystems: " \
 			$"Unmounting network block filesystems (retry): "
 	fi
   	if [ -n "$NFSMTAB" ]; then
-		__umount_loop '$3 ~ /^nfs/ && $3 != "nfsd" && $2 != "/" {print $2}' \
+		__umount_loop '$3 ~ /^nfs/ && $3 != "nfsd" && $2 != "/" && $2 !~ '$xtab' {print $2}' \
 			/proc/mounts \
 			$"Unmounting NFS filesystems: " \
 			$"Unmounting NFS filesystems (retry): " \
