--- netfs.orig	2008-06-05 21:23:18.945202828 +0200
+++ netfs	2008-06-05 21:35:51.682703206 +0200
@@ -30,7 +30,7 @@
 SMBMTAB=`LC_ALL=C awk '$3 == "smbfs" { print $2 }' /proc/mounts`
 CIFSMTAB=`LC_ALL=C awk '$3 == "cifs" { print $2 }' /proc/mounts`
 NCPMTAB=`LC_ALL=C awk '$3 == "ncpfs" { print $2 }' /proc/mounts`
-NETDEVMTAB=`LC_ALL=C awk '$4 ~ /_netdev/ && $2 != "/" { print $2 }' /etc/mtab`
+NETDEVMTAB=`LC_ALL=C awk '$4 ~ /_netdev/ && $2 != "/" && $2 != "/cdsl.local" { print $2 }' /etc/mtab`
 
 # See how we were called.
 case "$1" in
