--- netfs.orig	2009-02-04 10:35:06.000000000 +0100
+++ netfs	2009-02-04 10:35:57.000000000 +0100
@@ -13,6 +13,8 @@
 # Provides: $local_fs $remote_fs
 ### END INIT INFO
 
+# Patched for comoonics patch 1.1
+
 [ -f /etc/sysconfig/network ] || exit 0
 . /etc/init.d/functions
 . /etc/sysconfig/network
@@ -26,11 +28,11 @@
 NCPFSTAB=`LC_ALL=C awk '!/^#/ && $3 == "ncpfs" && $4 !~ /noauto/ { print $2 }' /etc/fstab`
 NETDEVFSTAB=`LC_ALL=C awk '!/^#/ && $4 ~/_netdev/ && $4 !~ /noauto/ { print $1 }' /etc/fstab`
 
-NFSMTAB=`LC_ALL=C awk '$3 ~ /^nfs/ && $3 != "nfsd" && $2 != "/" { print $2 }' /proc/mounts`
+NFSMTAB=`LC_ALL=C awk '$3 ~ /^nfs/ && $3 != "nfsd" && $2 != "/" && $2 != "/cdsl.local" { print $2 }' /proc/mounts`
 SMBMTAB=`LC_ALL=C awk '$3 == "smbfs" { print $2 }' /proc/mounts`
 CIFSMTAB=`LC_ALL=C awk '$3 == "cifs" { print $2 }' /proc/mounts`
 NCPMTAB=`LC_ALL=C awk '$3 == "ncpfs" { print $2 }' /proc/mounts`
-NETDEVMTAB=`LC_ALL=C awk '$4 ~ /_netdev/ && $2 != "/" { print $2 }' /etc/mtab`
+NETDEVMTAB=`LC_ALL=C awk '$4 ~ /_netdev/ && $2 != "/" && $2 != "/cdsl.local" { print $2 }' /etc/mtab`
 
 # See how we were called.
 case "$1" in
