--- network.orig	2009-06-04 20:16:22.000000000 +0200
+++ network	2009-06-04 20:15:50.000000000 +0200
@@ -23,6 +23,8 @@
 	. /etc/sysconfig/pcmcia
 fi
 
+# don't umount everything specified in xrootfs
+xrootfs=$(awk 'BEGIN{printf"(^"}{ gsub(/\//, "\\/"); printf $0"$|";}END{printf"^$)"}' /etc/xrootfs 2>/dev/null)
 
 # Check that networking is up.
 [ "${NETWORKING}" = "no" ] && exit 0
@@ -171,10 +173,10 @@
   stop)
   	# Don't shut the network down if root is on NFS or a network
 	# block device.
-        rootfs=$(awk '{ if ($1 !~ /^[ \t]*#/ && $2 == "/" && $3 != "rootfs") { print $3; }}' /proc/mounts)
+        rootfs=$(awk '{ if ($1 !~ /^rootfs/ && $1 !~ /^[ \t]*#/ && $2 == "/") { print $3; }}' /etc/mtab)
         rootopts=$(awk '{ if ($1 !~ /^[ \t]*#/ && $2 == "/") { print $4; }}' /etc/mtab)
 	
-	if [[ "$rootfs" =~ "^nfs" ]] || [[ "$rootopts" =~ "_netdev|_rnetdev" ]] ; then
+	if [[ "$rootfs" =~ "^nfs" ]]  || [[ "$rootfs" =~ $xrootfs ]] || [[ "$rootopts" =~ "_netdev|_rnetdev" ]] ; then
 		exit 1
 	fi
   
