--- netfs.orig	2008-06-23 16:09:44.000000000 +0200
+++ netfs	2008-06-23 16:08:05.000000000 +0200
@@ -13,6 +13,8 @@
 # Provides: $local_fs $remote_fs
 ### END INIT INFO
 
+# Patched for comoonics patch 1.0
+
 [ -f /etc/sysconfig/network ] || exit 0
 . /etc/init.d/functions
 . /etc/sysconfig/network
@@ -30,7 +32,7 @@
 SMBMTAB=`LC_ALL=C awk '$3 == "smbfs" { print $2 }' /proc/mounts`
 CIFSMTAB=`LC_ALL=C awk '$3 == "cifs" { print $2 }' /proc/mounts`
 NCPMTAB=`LC_ALL=C awk '$3 == "ncpfs" { print $2 }' /proc/mounts`
-NETDEVMTAB=`LC_ALL=C awk '$4 ~ /_netdev/ && $2 != "/" { print $2 }' /etc/mtab`
+NETDEVMTAB=`LC_ALL=C awk '$4 ~ /_netdev/ && $2 != "/" && $2 != "/cdsl.local" { print $2 }' /etc/mtab`
 
 # See how we were called.
 case "$1" in
